### PR TITLE
XCM: `ExpectTransactStatus` instruction

### PR DESCRIPTION
**Pull request by gavofyork**
cumulus companion: paritytech/cumulus#2112

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2282897

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2282965

**Comment by gavofyork** [comment]
@KiChjang Could you look into the cumulus companion?

**Review by KiChjang — APPROVED** [review]

File: xcm/src/v3/mod.rs [hunk 1]
```diff
@@ -806,30 +806,39 @@ pub enum Instruction<Call> {
...
 	/// - `ExpectationFalse`: If the value of the Error Register is not equal to the parameter.
 	ExpectError(Option<(u32, Error)>),
 
+	/// Ensure that the Transact StatusError Register equals some given value and throw an error if
+	/// not.
+	///
+	/// Kind: *Instruction*
+	///
+	/// Errors:
+	/// - `ExpectationFalse`: If the value of the Error Register is not equal to the parameter.
```
Comment (KiChjang) [inline xcm/src/v3/mod.rs:839]:
```suggestion
	/// - `ExpectationFalse`: If the value of the Transact Status Register is not equal to the parameter.
```

**Review by ggwpez — APPROVED** [review]

**Comment by KiChjang** [comment]
bot merge

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-38/2122/1

**Pull request merged**